fastlane/lib/fastlane/actions/app_store_build_number.rb

```diff
@@ -32,36 +32,42 @@ def self.get_build_number(params)
           build_nr = app.live_version.current_build_number
 
           UI.message("Latest upload for live-version #{app.live_version.version} is build: #{build_nr}")
+
+          return build_nr
         else
           version_number = params[:version]
 
-          # Get version number from latest pre-release if no version number given
+          # Used for determining how to display the optional version number
+          version_number_message = version_number.nil? ? "any version" : "version #{version_number}"
+
+          # Create filter for get_builds with optional version number
+          filter = { app: app.apple_id }
           if version_number
-            UI.message("Fetching the latest build number for version #{version_number}")
-
-            # Get latest build for version number
-            client = Spaceship::ConnectAPI::Base.client
-            build = client.get_builds(filter: { app: app.apple_id, "preReleaseVersion.version" => version_number }, sort: "-version", limit: 1).first
-            if build
-              build_nr = build.version
-            else
-              UI.important("Could not find a build number for version #{version_number} on App Store Connect")
-            end
+            filter["preReleaseVersion.version"] = version_number
+          end
+
+          UI.message("Fetching the latest build number for #{version_number_message}")
+
+          # Get latest build for optional version number and return build number if found
+          client = Spaceship::ConnectAPI::Base.client
+          build = client.get_builds(filter: filter, sort: "-version", includes: "preReleaseVersion", limit: 1).first
+          if build
+            build_nr = build.version
+            UI.message("Latest upload for version #{build.app_version} is build: #{build_nr}")
+            return build_nr
```

Comment:
This lookup feels like it could be moved into a little private method. It seems useful and self-contained.